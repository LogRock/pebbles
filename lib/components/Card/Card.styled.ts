import styled from "styled-components";

import { CardProps } from "./Card.types";

export const CardWrapper = styled.div<
  Pick<CardProps, "bgColor" | "borderRadius" | "padding" | "shadow">
>`
  display: flex;
  flex-direction: column;
  width: max-content;
  padding: ${({ padding }) => (padding ? `${padding}px` : "0px")};

  border-radius: ${({ borderRadius }) =>
    borderRadius ? `${borderRadius}px` : "none"};

  background-color: ${({ bgColor }) => bgColor};
  box-shadow: ${({ theme, shadow }) => (shadow ? theme.shadows.small : "none")};

  font-family: ${({ theme }) => theme.card.fontFamily};
  font-size: ${({ theme }) => theme.card.fontSize};
  font-weight: ${({ theme }) => theme.card.fontWeight};
`;
