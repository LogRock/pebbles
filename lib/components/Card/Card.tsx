import styled, { css } from "styled-components";
import { Shadows, Spacings } from "../../types/theme";

export interface CardProps {
  shadow?: keyof Shadows;
  padding?: keyof Spacings;
  radius?: keyof Spacings;
  noPadding?: boolean;
}

const Card = styled.div<CardProps>`
  padding: ${({ theme, padding }) => theme.spacings[padding || "md"]};

  border-radius: ${({ theme, radius }) => theme.spacings[radius || "sm"]};

  background: white;
  box-shadow: ${({ theme, shadow }) => theme.shadows[shadow || "small"]};

  ${({ noPadding }) =>
    noPadding &&
    css`
      padding: 0;
      overflow: hidden;
    `}
`;

export default Card;
