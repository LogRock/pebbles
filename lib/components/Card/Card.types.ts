export interface CardProps {
  bgColor?: string;
  padding?: number;
  borderRadius?: number;
  shadow?: boolean;
}
