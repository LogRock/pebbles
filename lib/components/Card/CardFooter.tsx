import React, { FC } from "react";

/**
 * @deprecated in favor of Card2
 */
const CardFooter: FC = ({ children, ...props }) => {
  return <div {...props}>{children}</div>;
};

/**
 * @deprecated in favor of Card2
 */
export default CardFooter;
