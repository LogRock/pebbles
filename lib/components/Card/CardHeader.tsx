import React, { FC } from "react";

const CardHeader: FC = ({ children, ...props }) => {
  return <div {...props}>{children}</div>;
};

export default CardHeader;
