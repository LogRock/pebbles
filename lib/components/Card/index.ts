/**
 * @deprecated in favor of Card2
 */
export { default } from "./Card";

/**
 * @deprecated in favor of Card2
 */
export type { CardProps } from "./Card.types";
