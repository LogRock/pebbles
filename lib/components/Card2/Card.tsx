import styled from "styled-components";
import { Shadows, Spacings } from "../../types/theme";

export interface CardProps {
  shadow?: keyof Shadows;
  padding?: keyof Spacings;
  radius?: keyof Spacings;
}

const Card2 = styled.div<CardProps>`
  padding: ${({ theme, padding }) => theme.spacings[padding || "md"]};

  border-radius: ${({ theme, radius }) => theme.spacings[radius || "sm"]};

  background: ${({ theme }) => theme.colors.shades[0]};
  box-shadow: ${({ theme, shadow }) => theme.shadows[shadow || "small"]};
`;

export default Card2;
