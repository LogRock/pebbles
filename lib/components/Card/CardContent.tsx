import React, { FC } from "react";

import { CardProps } from "./Card.types";

const CardContent: FC<CardProps> = ({ children, ...props }) => {
  return <div {...props}>{children}</div>;
};

export default CardContent;
