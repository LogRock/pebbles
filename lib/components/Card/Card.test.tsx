import React from "react";
import { render } from "../../utils/test-utils";
import Card from "./Card";

describe("Card", () => {
  it("renders with default props", () => {
    const { container } = render(<Card />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom props", () => {
    const { container } = render(
      <Card padding="huge" shadow="upXxLarge" radius="xxhuge" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
