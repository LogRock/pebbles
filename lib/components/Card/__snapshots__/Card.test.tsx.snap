// Vitest Snapshot v1

exports[`Card > renders with custom props 1`] = `
<div
  class="sc-bcXHqe kmkxeo"
  radius="xxhuge"
/>
`;

exports[`Card > renders with default props 1`] = `
<div
  class="sc-bcXHqe dWgcFr"
/>
`;
